sql/table.sql

```diff
@@ -2197,6 +2200,7 @@ CREATE TABLE biotype (
   description     TEXT,
   biotype_group   ENUM('coding','pseudogene','snoncoding','lnoncoding','mnoncoding','LRG','undefined','no_group') DEFAULT NULL,
   so_acc          VARCHAR(64),
+  so_term         VARCHAR(1023),
```

Comment:
Why are you using such a large number for the varchar? Even though the used space is variable MySQL will have allotted memory units reserved for the storage of the string, which will be much larger in this case. This will reflect badly when sorting or temp tables are required.

Might be worth analysing the required size that is need and restricting what you are storing. Is worth adding a test in the code to ensure that there is not truncation of the inserted text at a later date.

Reply:
That data is present and extracted from the ensembl_ontology_<version> database, term table, name column, which is a VARCHAR(1023).

It could be a TEXT instead, but it's usually better to be a VARCHAR if within it's limits and as you say it will only take as much space as it needs...
The usage is expected to be through the Biotype object, which is a select of a specific biotype name and object_type combination.

Reply:
`TEXT` would be the wrong way, if the object does not need to be up to 1023 then limiting down to a smaller number will help MySQL allot the correct memory size.

Reply:
As I said, the source data is 1023 so it can theoretically be that large.
I do think it probably won't be, but looking at the sequenceontology website I can't find any real numbers for how large a term can be.
The only way to play safe is to make it as large as the source data can be.
The table is however quite small, and it's usage limited (and mostly as the select I mentioned).